internal/operator-controller/authorization/rbac.go

```diff
@@ -0,0 +1,464 @@
+package authorization
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"sort"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	rbacinternal "k8s.io/kubernetes/pkg/apis/rbac"
+	rbacv1helpers "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	rbacregistry "k8s.io/kubernetes/pkg/registry/rbac"
+	"k8s.io/kubernetes/pkg/registry/rbac/validation"
+	rbac "k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PreAuthorizer interface {
```

Comment:
we could declare this interface on the applier/helm side to decouple it completely from this package as a dependency, but I saw that we're usually declaring interfaces on the producer side, probably because those are then reused, so seems fine as is now too

Reply:
We're going to add unit tests based on your general comment I think. If so it might one more reason to change where the interface is made, could help with testing...

Reply:
these tests were added, marking resolved